dali/kernels/reduce/reduce_cpu.h

```diff
@@ -236,13 +237,15 @@ struct ReduceBaseCPU {
 
   void InitAxes(span<const int> _axes) {
     for (int axis : _axes) {
-      if (axis < 0 || axis >= ndim()) {
-        throw std::range_error(make_string("Axis index out of range: ", axis, " not in 0..",
-                               ndim()-1));
+      if (axis < -ndim() || axis >= ndim()) {
+        throw std::range_error(
+            make_string("Axis index out of range: ", axis, " not in ", -ndim(), "..", ndim() - 1));
```

Comment:
How about applying mathematical range notation?
```suggestion
            make_string("Axis index out of range: ", axis, " not in [", -ndim(), ", ", ndim() - 1, "]"));
```